tests/functional/tests/imu_kit/suite_imu_kit.cpp

```diff
@@ -0,0 +1,130 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "rtos/ThisThread.h"
+
+#include "CoreAccelerometer.h"
+#include "CoreGyroscope.h"
+#include "CoreI2C.h"
+#include "CoreLSM6DSOX.h"
+#include "EventLoopKit.h"
+#include "IMUKit.h"
+#include "tests/config.h"
+
+using namespace leka;
+using namespace std::chrono;
+using namespace boost::ut;
+using namespace boost::ut::bdd;
+
+suite suite_lsm6dsox = [] {
+	constexpr auto default_min_bound_pitch = -5._f;
+	constexpr auto default_max_bound_pitch = 5._f;
+	constexpr auto default_min_bound_roll  = -5._f;
+	constexpr auto default_max_bound_roll  = 5._f;
+	constexpr auto default_min_bound_yaw   = 170._f;
+	constexpr auto default_max_bound_yaw   = 190._f;
+	constexpr auto measures_number		   = 100;
+	constexpr auto maximal_yaw_drift	   = 0.1F;
+
+	auto i2c		= CoreI2C(PinName::SENSOR_IMU_TH_I2C_SDA, PinName::SENSOR_IMU_TH_I2C_SCL);
+	auto lsm6dsox	= CoreLSM6DSOX {i2c};
+	auto accel		= CoreAccelerometer {lsm6dsox};
+	auto gyro		= CoreGyroscope {lsm6dsox};
+	auto event_loop = EventLoopKit {};
+
+	auto imukit = IMUKit {event_loop, accel, gyro};
+
+	lsm6dsox.init();
+
+	imukit.start();
+
+	"Initialization"_test = [&] {
+		expect(neq(&imukit, nullptr));
+		imukit.init();
+	};
+
+	"Start"_test = [&] {};
+
+	"get angles"_test = [&] {
+		imukit.start();
```

Comment:
These tests are somehow not homogeneous, there are initialization, then some test then a scenario (that does not restart imukit). 

There is even a `start` before an `init` for imukit

Maybe you can move the `"get angles"` test in a scenario where you initialize imukit and start it, and `initialize`+`start` instead of `reset` in `scenario("imu - drift")`